### PR TITLE
[CARBONDATA-3119] Fixing the getOrCreateCarbonSession method parameter to an empty string

### PR DESCRIPTION
**Pull request by zygitup**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 1]
```diff
@@ -248,7 +248,7 @@ object CarbonSession {
 
         session = new CarbonSession(sparkContext, None, !enableInMemCatlog)
         val carbonProperties = CarbonProperties.getInstance()
-        if (storePath != null) {
+        if (storePath != null && !storePath.trim.isEmpty) {
```
Comment (qiuchenjian) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: why not use StringUtils or other String tools

**Comment by chenliang613** [comment]
add to whitelist

**Comment by chenliang613** [comment]
the pr title is not correct, the format should be : [JIRA NUMBER] PR description

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1574/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1786/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9833/

**Comment by xubo245** [comment]
Please optimize the title, for example: [CARBONDATA-3119]Fix the error of getOrCreateCarbonSession method parameter is an empty string
Please optimize the PR content, keep the checklist format and finish the checklist

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala (hunk 1 above)
Comment (qiuchenjian) [reply integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: such as StringUtils.isNotEmpty

Comment (xubo245) [reply integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: StringUtils.isNotEmpty shouldn't support " "

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 2]
```diff
@@ -180,7 +180,7 @@ object CarbonSession {
       val userSuppliedContext: Option[SparkContext] =
         getValue("userSuppliedContext", builder).asInstanceOf[Option[SparkContext]]
 
-      if (metaStorePath != null) {
+      if (metaStorePath != null && !metaStorePath.trim.isEmpty) {
```
Comment (xubo245) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: Can you add some test case to test it?

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala (hunk 1 above)
Comment (xubo245) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: Can you add some test case to test it?

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala (hunk 2 above)
Comment (zzcclp) [reply integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: Use StringUtils.isNotBlank.

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala (hunk 1 above)
Comment (zzcclp) [reply integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: Use StringUtils.isNotBlank.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1595/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9855/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1807/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1601/

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 3]
```diff
@@ -180,7 +183,7 @@ object CarbonSession {
       val userSuppliedContext: Option[SparkContext] =
         getValue("userSuppliedContext", builder).asInstanceOf[Option[SparkContext]]
 
-      if (metaStorePath != null) {
+      if (metaStorePath != null && StringUtils.isNotBlank(metaStorePath)) {
```
Comment (xubo245) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: StringUtils.isNotBlank will judge null

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 4]
```diff
@@ -37,6 +38,8 @@ import org.apache.carbondata.core.constants.CarbonCommonConstants
 import org.apache.carbondata.core.util.{CarbonProperties, CarbonSessionInfo, ThreadLocalSessionInfo}
 import org.apache.carbondata.streaming.CarbonStreamingQueryListener
 
+
+
```
Comment (zzcclp) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: remove these two lines.

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 5]
```diff
@@ -248,7 +251,7 @@ object CarbonSession {
 
         session = new CarbonSession(sparkContext, None, !enableInMemCatlog)
         val carbonProperties = CarbonProperties.getInstance()
-        if (storePath != null) {
+        if (storePath != null && StringUtils.isNotBlank(storePath)) {
```
Comment (zzcclp) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: remove 'storePath != null', StringUtils.isNotBlank already includes this judgement.

**Comment by zzcclp** [comment]
If possible, it better add a test case for this.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9861/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1812/

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala (hunk 4 above)
Comment (zygitup) [reply integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: ok.Has been removed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1603/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9863/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1814/

**Comment by xuchuanyin** [comment]
LGTM

**Comment by xuchuanyin** [comment]
LGTM

**Comment by xuchuanyin** [comment]
LGTM

**Comment by zzcclp** [comment]
LGTM

**Pull request closed without merging**